src/Raven.Server/Documents/Indexes/IndexNameFinder.cs

```diff
@@ -37,13 +37,13 @@ private static string FindName(string[] collections, IReadOnlyCollection<IndexFi
             
             var combinedFields = string.Join("And", fields.Select(x => IndexField.ReplaceInvalidCharactersInFieldName(x.Name)).OrderBy(x => x));
```

Comment:
We are going to be doing this at least once a query, so this is going to be called a lot, and we are doing a lot of allocations here.
We need to see how we can reduce as much as possible the number of allocations made.